lib/connect.ts

```diff
@@ -65,1354 +69,648 @@ import utils from "./utils";
+            }
         }
+    }
 
-        return data;
+    return data;
+}
+
+// Format response ex. datetime string to date
+function formatResponse(data: AnyObject): AnyObject {
+    if (!data.data || typeof data.data !== 'object') return data;
+    let list: any = [];
+    if (data.data instanceof Array) {
+        list = data.data;
+    } else {
+        list = [data.data]
     }
 
-    function transformMFInstrumentsResponse(data, headers) {
-        // Parse CSV responses
-        if (headers["content-type"] === "text/csv") {
-            const parsedData = csvParse.parse(data, {"header": true}).data;
-            for (const item of parsedData as any) {
-                item["minimum_purchase_amount"] = parseFloat(item["minimum_purchase_amount"]);
-                item["purchase_amount_multiplier"] = parseFloat(item["purchase_amount_multiplier"]);
-                item["minimum_additional_purchase_amount"] = parseFloat(item["minimum_additional_purchase_amount"]);
-                item["redemption_quantity_multiplier"] = parseFloat(item["redemption_quantity_multiplier"]);
-                item["minimum_additional_purchase_amount"] = parseFloat(item["minimum_additional_purchase_amount"]);
-                item["last_price"] = parseFloat(item["last_price"]);
-                item["purchase_allowed"] = Boolean(parseInt(item["purchase_allowed"]));
-                item["redemption_allowed"] = Boolean(parseInt(item["redemption_allowed"]));
-
-                if (item["last_price_date"] && item["last_price_date"].length === 10) {
-                    item["last_price_date"] = new Date(item["last_price_date"]);
-                }
-            }
+    const results: any[] = [];
+    const fields = ['order_timestamp', 'exchange_timestamp', 'created', 'last_instalment', 'fill_timestamp'];
 
-            return parsedData;
+    for (const item of list) {
+        for (const field of fields) {
+            if (item[field] && item[field].length === 19) {
+                item[field] = new Date(item[field]);
+            }
         }
 
-        return data;
+        results.push(item);
     }
 
-    function _get(route, params?, responseType?, responseTransformer?, isJSON = false) {
-        return request(route, "GET", params || {}, responseType, responseTransformer, isJSON);
+    if (data.data instanceof Array) {
+        data.data = results;
+    } else {
+        data.data = results[0];
     }
 
-    function _post(route, params, responseType?, responseTransformer?, isJSON = false, queryParams: any =null) {
-        return request(route, "POST", params || {}, responseType, responseTransformer, isJSON, queryParams);
-    }
+    return data;
+}
 
-    function _put(route, params, responseType?, responseTransformer?, isJSON = false, queryParams=null) {
-        return request(route, "PUT", params || {}, responseType, responseTransformer, isJSON, queryParams);
-    }
+function parseHistorical(jsonData: AnyObject): AnyObject {
```

Comment:
Better to make `parseHistorical` function private `private parseHistorical(jsonData: AnyObject): AnyObject`.